### PR TITLE
Create lighthouse ci workflow action

### PR DESCRIPTION
**Pull request by jlieb10**
## What does this change?
This will allow us to see the lighthouse scores of our builds to be aware if we are regressing before pushing to master

## Why?
We want to keep dotcom-rendering ultra performant and accessible

## Link to supporting Trello card
https://trello.com/c/Rgtlap7f/1582-implement-lighthouse-regression-testing-github-integration

### CODE REVIEW

**Comment by jfsoul** [comment]
@Jpsi you might be interested in this too

**Comment by jlieb10** [comment]
I need to figure out how to configure this to use the appropriate URL

**Comment by oliverlloyd** [comment]
> I need to figure out how to configure this to use the appropriate URL

This could be tricky. In dev mode you can make GET requests against DCR so it works as a webserver in the classic sense. But in production mode, the one you want to test here, it only accepts POSTS and depends on the existence of `frontend` to serve the page.

We get around this when running Cypress tests by testing in dev mode like this:

```
@NODE_ENV=development start-server-and-test 'node scripts/frontend/dev-server' 3030 'cypress run --spec "cypress/integration/e2e/**/*"'
```

This isn't great but works.

I wonder if we need to think about adding an endpoint to DCR to act as a webserver. It would be somewhat odd though.

1. User (CI test process) hits DCR GET endpoint with article path
2. DCR calls out to `frontend` to ask for the article
3. Frontend would call CAPI, make some changes, and then POST the result to DCR
4. DCR returns the rendered html to Frontend
5. Frontend returns the same back to the original request from DCR
6. DCR returns html to user

Hmmm, that's not confusing at all!

**Comment by jlieb10** [comment]
@oliverlloyd l'm a bit lost on that, to be honest. Is it something you'd be able to work on with me? Thanks a lot.

**Comment by oliverlloyd** [comment]
> @oliverlloyd l'm a bit lost on that, to be honest. Is it something you'd be able to work on with me? Thanks a lot.

Absolutely.

To be honest, it still confuses me sometimes but lets have a chat directly to lay it all out. @gtrufitt maybe you can join as well as I have a couple of questions myself.

**Comment by liywjl** [comment]
I think this is a common patter to have when you have a review apps that auto build on each PR. Maybe this is something we could explore? its a nice way of testing the PR without having to run it locally

**Comment by gtrufitt** [comment]
> @gtrufitt maybe you can join as well as I have a couple of questions myself.

Lets chat after standup!

File: makefile [hunk 1]
```diff
@@ -30,7 +30,7 @@ deploy:
 
 # prod #########################################
 
-build: clear clean-dist install
```
Comment (liywjl) [inline makefile:33]: why remove `clear`? this would cause us to bloat the `/dist` folder

**Comment by liywjl** [comment]
🎊 looks like lighthouse works on the CI!

**Comment by liywjl** [comment]
where is the output of lighthouse found?

**Comment by jlieb10** [comment]
@liywjl I am hoping to get it working first then worry about where the results go or setting a pass/fail threshold

File: makefile (hunk 1 above)
Comment (jlieb10) [reply makefile:33]: @liywjl it relies on the commands being run with a $TERM set, which is not the case for Github actions. Another option is set up a custom 'lighthouse build' command

**Comment by oliverlloyd** [comment]
Closing this for now. We can reopen if this is still needed

**Pull request closed without merging**